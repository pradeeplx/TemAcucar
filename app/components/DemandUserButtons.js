import React, { Component } from 'react'
import { Alert, View, Text } from 'react-native'
import Colors from "../Colors"
import BottomView from "./BottomView"
import BottomButton from "./BottomButton"

export default class DemandUserButtons extends Component {
  handleComplete() {
    Alert.alert(
      'Concluir pedido?',
      'Você tem certeza que quer concluir este pedido?',
      [{ text: 'Não', style: 'cancel' }, { text: 'Sim', onPress: () => {
        const { demand, onComplete } = this.props
        onComplete(demand)
      }}]
    )
  }

  handleCancel() {
    Alert.alert(
      'Cancelar pedido?',
      'Você tem certeza que quer cancelar este pedido?',
      [{ text: 'Não', style: 'cancel' }, { text: 'Sim', onPress: () => {
        const { demand, onCancel } = this.props
        onCancel(demand)
      }}]
    )
  }

  handleReactivate() {
    Alert.alert(
      'Reativar pedido?',
      'Você tem certeza que quer reativar este pedido?',
      [{ text: 'Não', style: 'cancel' }, { text: 'Sim', onPress: () => {
        const { demand, onReactivate } = this.props
        onReactivate(demand)
      }}]
    )
  }

  render() {
    const { demand, admin, onComplete, onCancel, onReactivate } = this.props
    const { state, completing, canceling, reactivating } = demand
    const canComplete = (onComplete && (state === 'notifying' || state === 'active'))
    const canCancel = (onCancel && (state === 'notifying' || state === 'active'  || (state === 'flagged' && admin)))
    const canReactivate = (onReactivate && (state === 'completed' || ((state === 'flagged' || state === 'canceled') && admin)))
    if (canReactivate && canCancel) {
      return (
        <BottomView>
          <BottomButton
            onPress={this.handleReactivate.bind(this)}
            isLoading={reactivating}
          >
            Reativar pedido
          </BottomButton>
          <BottomButton
            onPress={this.handleCancel.bind(this)}
            isLoading={canceling}
            secondary={true}
          >
            Cancelar pedido
          </BottomButton>
        </BottomView>
      )
    }
    if (canReactivate) {
      return (
        <BottomView>
          <BottomButton
            onPress={this.handleReactivate.bind(this)}
            isLoading={reactivating}
          >
            Reativar pedido
          </BottomButton>
        </BottomView>
      )
    }
    if (canComplete && canCancel) {
      return (
        <BottomView>
          <BottomButton
            onPress={this.handleComplete.bind(this)}
            isLoading={completing}
          >
            { admin ? 'Concluir pedido' : 'Já consegui' }
          </BottomButton>
          <BottomButton
            onPress={this.handleCancel.bind(this)}
            isLoading={canceling}
            secondary={true}
          >
            Cancelar pedido
          </BottomButton>
        </BottomView>
      )
    }
    return null
  }
}
