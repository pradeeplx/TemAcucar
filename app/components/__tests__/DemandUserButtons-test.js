import { View } from 'react-native'
import React from 'react'
import DemandUserButtons from '../DemandUserButtons'
import renderer from 'react-test-renderer'

const props = {
  demand: {
    state: 'active',
    completing: false,
    canceling: false,
    reactivating: false,
  },
  admin: true,
}

it('renders null when it has no callbacks', () => {
  const tree = renderer.create(
    <DemandUserButtons {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot()
});

it('renders correctly when canComplete and canCancel', () => {
  const tree = renderer.create(
    <DemandUserButtons {...props} onComplete={true} onCancel={true} />
  ).toJSON();
  expect(tree).toMatchSnapshot()
});

it('renders correctly when canComplete and canCancel for non admins', () => {
  const tree = renderer.create(
    <DemandUserButtons {...props} onComplete={true} onCancel={true} admin={false} />
  ).toJSON();
  expect(tree).toMatchSnapshot()
});

it('renders correctly when canReactivate', () => {
  const tree = renderer.create(
    <DemandUserButtons {...props} onReactivate={true} demand={{ ...props.demand, state: 'completed' }} />
  ).toJSON();
  expect(tree).toMatchSnapshot()
});

it('renders correctly when canReactivate and canCancel', () => {
  const tree = renderer.create(
    <DemandUserButtons {...props} onReactivate={true} onCancel={true} demand={{ ...props.demand, state: 'flagged' }} />
  ).toJSON();
  expect(tree).toMatchSnapshot()
});
