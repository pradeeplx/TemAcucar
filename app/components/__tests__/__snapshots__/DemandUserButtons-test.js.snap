exports[`test renders correctly when canComplete and canCancel 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "bottom": 0,
        "flexDirection": "row",
        "left": 0,
        "position": "absolute",
        "right": 0,
      },
      undefined,
    ]
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#FF7F85",
        "borderRadius": 0,
        "borderWidth": 0,
        "flex": 1,
        "flexDirection": "row",
        "height": 105,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": 0,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 18,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 21,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
            undefined,
          ],
        ]
      }>
      Concluir pedido
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#E6D3B7",
        "borderRadius": 0,
        "borderWidth": 0,
        "flex": 1,
        "flexDirection": "row",
        "height": 105,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": 0,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 18,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 21,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
            undefined,
          ],
        ]
      }>
      Cancelar pedido
    </Text>
  </View>
</View>
`;

exports[`test renders correctly when canComplete and canCancel for non admins 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "bottom": 0,
        "flexDirection": "row",
        "left": 0,
        "position": "absolute",
        "right": 0,
      },
      undefined,
    ]
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#FF7F85",
        "borderRadius": 0,
        "borderWidth": 0,
        "flex": 1,
        "flexDirection": "row",
        "height": 105,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": 0,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 18,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 21,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
            undefined,
          ],
        ]
      }>
      JÃ¡ consegui
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#E6D3B7",
        "borderRadius": 0,
        "borderWidth": 0,
        "flex": 1,
        "flexDirection": "row",
        "height": 105,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": 0,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 18,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 21,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
            undefined,
          ],
        ]
      }>
      Cancelar pedido
    </Text>
  </View>
</View>
`;

exports[`test renders correctly when canReactivate 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "bottom": 0,
        "flexDirection": "row",
        "left": 0,
        "position": "absolute",
        "right": 0,
      },
      undefined,
    ]
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#FF7F85",
        "borderRadius": 0,
        "borderWidth": 0,
        "flex": 1,
        "flexDirection": "row",
        "height": 105,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": 0,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 18,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 21,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
            undefined,
          ],
        ]
      }>
      Reativar pedido
    </Text>
  </View>
</View>
`;

exports[`test renders correctly when canReactivate and canCancel 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "bottom": 0,
        "flexDirection": "row",
        "left": 0,
        "position": "absolute",
        "right": 0,
      },
      undefined,
    ]
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#FF7F85",
        "borderRadius": 0,
        "borderWidth": 0,
        "flex": 1,
        "flexDirection": "row",
        "height": 105,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": 0,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 18,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 21,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
            undefined,
          ],
        ]
      }>
      Reativar pedido
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#E6D3B7",
        "borderRadius": 0,
        "borderWidth": 0,
        "flex": 1,
        "flexDirection": "row",
        "height": 105,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 0,
        "paddingVertical": 0,
        "width": 0,
      }
    }
    testID={undefined}>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flex": 1,
            "fontSize": 18,
            "textAlign": "center",
          },
          Array [
            Object {
              "color": "#FFFFFF",
              "fontSize": 21,
              "textAlign": "center",
            },
            Object {
              "fontFamily": "Avenir-Black",
            },
            undefined,
          ],
        ]
      }>
      Cancelar pedido
    </Text>
  </View>
</View>
`;

exports[`test renders null when it has no callbacks 1`] = `null`;
